File: leetcode/283.move-zeroes.go
package main

/*
 * @lc app=leetcode id=283 lang=golang
 *
 * [283] Move Zeroes
 */

// @lc code=start
func moveZeroes(nums []int) {
	curr := 0
	for i := 0; i < len(nums); i++ {
		if nums[i] != 0 {
			nums[curr] = nums[i]
			curr++
		}
	}
	for ; curr < len(nums); curr++ {
		nums[curr] = 0
	}
}

// @lc code=end

File: leetcode/283.move-zeroes.cpp
/*
 * @lc app=leetcode id=283 lang=cpp
 *
 * [283] Move Zeroes
 */

// @lc code=start
class Solution {
public:
    void moveZeroes(vector<int>& nums) {
        int i = 0;
        for (int j = 0; j < nums.size(); j++)
            if (nums[j] != 0)
                nums[i++] = nums[j];
        fill(nums.begin() + i, nums.end(), 0);
    }
};
// @lc code=end

